ocaml/xapi/xapi_host.ml

```diff
@@ -66,6 +66,19 @@ let assert_safe_to_reenable ~__context ~self =
     List.iter (fun self -> Xapi_pif.abort_if_network_attached_to_protected_vms ~__context ~self) unplugged_pifs;
   end
 
+(* The maximum pool size allowed must be restricted to 3 hosts for the pool which does not have Pool_size feature *)
```

Comment:
Or `must be restricted to [Xapi_globs.restricted_pool_size] hosts`, in case we change that variable